src/Features/Core/Portable/InitializeParameter/AbstractAddParameterCheckCodeRefactoringProvider.cs

```diff
@@ -144,6 +139,53 @@ internal abstract partial class AbstractAddParameterCheckCodeRefactoringProvider
 
         protected abstract bool CanOffer(SyntaxNode body);
 
+        private async Task<Document> UpdateDocumetForRefactoring(
+            Document document,
+            SyntaxNode functionDeclaration,
+            IBlockOperation blockStatementOpt,
+            List<int> listOfParametersOrdinals,
+            int position,
+            CancellationToken cancellationToken)
+        {
+
+            foreach (var index in listOfParametersOrdinals)
+            {
+                // Updates functionDeclaration and uses it to get the first valid ParameterNode using the ordinals (index).
+                var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                var token = root.FindToken(position);
+                var firstparameterNode = GetParameterNode(token, position);
+                functionDeclaration = firstparameterNode.FirstAncestorOrSelf<SyntaxNode>(IsFunctionDeclaration);
+
+                var currentNode = GetParameterNodeAtIndex(functionDeclaration, index);
+                if (currentNode == null)
+                {
+                    continue;
+                }
```

Comment:
this approach is *very* prone to failure.  it basically depends on absolutely no later code doing anything that might cause this lookup to fail.  We don't need this.  We can have the refactoring, it can grab all the parameters to add checks for, then it can produce all the checking-statements it needs, then it can add all those checking-statements at once to the body of the method.

Reply:
What do you mean by failure? If there's anything wrong with that parameter, it would just skip it and offer to add checks for the valid ones.

Reply:
>What do you mean by failure? 

You're mutating code and then trying to figure out what's going on post-edit.  That's "prone to failure" because it's not clear that your later work won't b thrown off by your previous edits.  

> If there's anything wrong with that parameter, it would just skip it and offer to add checks for the valid ones.

Right.  But that's just patching up a bad situation.  my point is that we can avoid the bad situation in the first place.  INstead of adding the null checks one-at-a-time and recomputing everything after each time you add, we can (and should) just add all the null checks all at once.  This can't fail and there would be no need to do things like "skip" any parameters then. :)

Thanks!